src/IAPWS_WaterProperties.f90

```diff
@@ -0,0 +1,4258 @@
+    ! to avoid negative roots outside of range of validity
+    IF (z8 < 0.0_SRK) THEN
+      z4 = -1.0_SRK*sqrt(-1.0_SRK*z8)
+      z2 = -1.0_SRK*sqrt(-1.0_SRK*z4)
+      z1 = -1.0_SRK*sqrt(-1.0_SRK*z2)
+    ELSE
+      z4 = sqrt(z8)
+      z2 = sqrt(z4)
+      z1 = sqrt(z2)
+    ENDIF
+    vpt3n = pr*e03 + tr*e01 + t2*(e02 + t2*t2*p5*e04) &
+            + z1*(t2*e05 &
+              + z1*(pr*e06 + t2*p5*e07 &
+                + z1*(tr*p4*e11 + t3*(e08 + tr*p2*pr*e10 + t2*e09) &
+                  + z1*(p5*e15 + t3*(e12 + tr*e13 + t3*e14) &
+                    + z1*(p4*(e16 + tr*e17) &
+                      + z1*(e18 + tr*(e19 + pr*e21 + p5*e23 + tr*(e20 + p2*e22)) &
+                        + z2*(e24 + pr*(e25 + pr*e27 + p2*e29 + tr*(pr*e28 + p2*e30 + tr*(e26 + p2*e31 &
+                                                                                      ))) &
+                          + z2*(pr*e34 + tr*(e32 + p5*pr*e36 + tr*(p4*e35 + t3*e33)) &
+                            + z2*(e37 + p2*e39 + p4*e40 + tr*pr*(e38 + tr*p5*e41))))))))))
+    vest3 = vpt3n + vc
+  ELSEIF (ireg3 == 6) THEN
+    pr = p - PCrit
+    p2 = pr*pr
+    tr = t - Tcrit
+    z4 = (0.1_SRK*pr - dtdp)*pr + (1.0_SRK + 0.002_SRK*tr)*tr
+    ! to avoid negative roots outside of range of validity
+    IF (z4 < 0.0_SRK) THEN
+      z2 = -1.0_SRK*sqrt(-1.0_SRK*z4)
+      z1 = -1.0_SRK*sqrt(-1.0_SRK*z2)
+    ELSE
+      z2 = sqrt(z4)
+      z1 = sqrt(z2)
+    ENDIF
+    z3 = z1*z2
+    vpt3n = z3*(f08 + z2*(f12 + p2*f14 + z1*f15)) &
+            + tr*(f01 + z1*(f04 + z1*(f06 + z1*(f09 + z2*(f13 + z1*p2*pr*f17)))) &
+              + tr*(f02 + z1*(f05 + z3*f11) &
+                + tr*(f03 + z3*f10 &
+                  + tr*z2*(f07 + z4*p2*f16))))
+    vest3 = vpt3n + vc
+    ! region 3g (40 MPa<p<100 MPa)
+  ELSEIF (ireg3 == 7) THEN
+    pr = p*(1.0_SRK/103.0_SRK) - 1.0_SRK
+    p2 = pr*pr
+    p4 = p2*p2
+    tr = t*(1.0_SRK/600.0_SRK) - 1.0_SRK
+    t2 = tr*tr
+    t3 = t2*tr
+    t4 = t2*t2
+    t6 = t3*t3
+    p2t2 = p2*t2
+    vpt3n = g01 + tr*(g02 + t2*(g03 + t3*g04)) &
+            + pr*(g05 + t2*g06 &
+              + pr*(tr*t4*g07 &
+                + pr*(tr*(g08 + t3*(g09 + t4*g10)) &
+                  + p2*(g11 &
+                    + p2*pr*(t6*g12 &
+                      + p2*t4*(t6*g13 &
+                        + p4*(g14 &
+                          + p2*t6*(t6*t6*g15 &
+                            + p4*p4*(g16 &
+                              + p2t2*(g17 &
+                                + p2t2*(g18 &
+                                  + p2t2*(g19 + p2t2*g20))))))))))))
+    vest3 = vpt3n
+  ENDIF
+ENDFUNCTION vest3
+!
+!-------------------------------------------------------------------------------
+!> Returns the specific volume of the phase.  Works for saturated liquid or
+!> vapor.
+!> @param P    The saturation pressure
+!> @param T    The saturation temperature
+!> @param DEST The density of the fluid [kg/m**3]
+!> @param EPS  TODO - what is EPS
```

Comment:
Through pure luck, happened to notice this... would be good to fill in the TODO if possible

Reply:
if I knew what it was... I'd fill it in :)